### PR TITLE
Setup basic typescript support

### PR DESCRIPTION
**Pull request by sjaakiejj**
Authzed node internally makes use of types, but then wraps them in javascript, destroying the types and making them unavailable to the client. With this PR we correct this behaviour and make the types available to developers using this package.

### CODE REVIEW

**Comment by bracki** [comment]
Have you pushed this to npm somewhere already so it can be tried out?

**Comment by sjaakiejj** [comment]
> How do you use `NewClient` then? It is of type `{}`. Do you refine it further when using it?

Because of the use of `Proxy`, a type for NewClient is not generated. For now we're casting it to 'any'. This PR is more about exporting the types that are already there but simply not exposed, which covers most of the more complex functionality.

**Comment by sjaakiejj** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by sjaakiejj** [comment]
recheck

File: README.md [hunk 1]
```diff
@@ -57,37 +57,43 @@ Everything required to connect and make API calls is located in a module respect
 You will have to provide a your own API Token from the [Authzed dashboard] in place of `t_your_token_here_1234567deadbeef` in the following example:
 
 ```js
-const authzed = require('@authzed/authzed-node/v0');
+import { v1 } from '@authzed/authzed-node';
```
Comment (samkim) [inline README.md:60]: Can you add a link for the v1 api to the list under "Supported client API versions" above? The link can point to `https://docs.authzed.com/reference/api#authzedapiv1`

**Review by samkim — COMMENTED** [review]
Thanks for adding the spicedb action. Just a minor readme update left.

Pull and was able to build and run on an M1 mac but had to build grpc-tools for x64. 
`yarn add grpc-tools --ignore-scripts `
`node_modules/.bin/node-pre-gyp install --target_arch=x64 `
Leaving a note here in case others run into this.

File: README.md (hunk 1 above)
Comment (sjaakiejj) [reply README.md:61]: Done

**Review by samkim — APPROVED** [review]
LGTM!

**Comment by samkim** [comment]
Btw I'll merge and cut a new release of the client tomorrow.

**Pull request merged**